File: spring-api/src/main/java/com/datatecsolution/eprofe/spring_api/model/Acumulativo.java
package com.datatecsolution.eprofe.spring_api.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDate;
import java.time.LocalDateTime;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Entity
@Table(name = "acumulativos")
@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Acumulativo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "seccion_id")
    private Seccion seccion;

    private String descripcion;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "tipo_acumulativo_id")
    private TipoAcumulativo tipoAcumulativo;

    private LocalDate fecha;

    private Integer parcial;

    private Double valor;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "asignatura_id")
    private Asignatura asignatura;

    @Column(name = "movil_id")
    private Integer movilId;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: mobile-app/model/Acumulativo.ts
import { Model } from '@nozbe/watermelondb'
import { field, date, relation, readonly } from '@nozbe/watermelondb/decorators'
import Seccion from './Seccion'
import Asignatura from './Asignatura'

export default class Acumulativo extends Model {
    static table = 'acumulativos'

    @field('descripcion') descripcion: string
    @field('valor') valor: number
    @field('parcial') parcial: number
    @field('fecha') fecha: string
    @field('uploaded') uploaded: boolean

    @readonly @date('created_at') createdAt: number
    @readonly @date('updated_at') updatedAt: number

    @relation('secciones', 'seccion_id') seccion: Seccion
    @relation('asignaturas', 'asignatura_id') asignatura: Asignatura
    @relation('tipo_acumulativos', 'tipo_acumulativo_id') tipoAcumulativo: any
}
